client/lcd/lcd_test.go

```diff
@@ -684,6 +693,48 @@ func doUnbond(t *testing.T, port, seed, name, password string, delegatorAddr, va
 	return results[0]
 }
 
+func doBeginRedelegation(t *testing.T, port, seed, name, password string,
+	delegatorAddr, validatorSrcAddr, validatorDstAddr sdk.Address) (resultTx ctypes.ResultBroadcastTxCommit) {
+
+	// get the account to get the sequence
+	acc := getAccount(t, port, delegatorAddr)
+	accnum := acc.GetAccountNumber()
+	sequence := acc.GetSequence()
+
+	delegatorAddrBech := sdk.MustBech32ifyAcc(delegatorAddr)
+	validatorSrcAddrBech := sdk.MustBech32ifyVal(validatorSrcAddr)
+	validatorDstAddrBech := sdk.MustBech32ifyVal(validatorDstAddr)
+
+	// send
+	jsonStr := []byte(fmt.Sprintf(`{
```

Comment:
Here might be where you want to pass `chain_id`, and set `ctx.ChainID = body.ChainID` in the handler